components/ScanNewSecret.js

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useReducer } from 'react'
+import { View, StyleSheet, Button } from 'react-native'
+import { BarCodeScanner } from 'expo-barcode-scanner'
+import { useNavigation } from '@react-navigation/native'
+
+import { useAuthenticationContext } from '../context/authentication'
+import { useSecretsContext } from '../context/secrets'
+import { parse } from '../lib/qrParser'
+import routes from '../lib/routeDefinitions'
+
+import { BodyText } from './typography'
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    flexDirection: 'column',
+    justifyContent: 'center',
+  },
+})
+
+const UI_STRINGS = {
+  undetermined: 'Requesting for camera permission',
+  denied: 'No access to camera',
+  active: 'Tap to Scan Again',
+}
+
+const PERMISSION_STATES = {
+  undetermined: 'undetermined',
+  granted: 'granted',
+  denied: 'denied',
+}
+
+const SCAN_STATES = {
+  none: 'none',
+  active: 'active',
+}
+
+const ACTION_TYPES = {
+  SET_SCAN: 'SET_SCAN',
+  SET_PERMISSION: 'SET_PERMISSION',
+}
+
+const initialState = {
+  permission: PERMISSION_STATES.undetermined,
+  scan: SCAN_STATES.none,
+}
+
+function scanReducer(state, action) {
+  switch (action.type) {
+    case ACTION_TYPES.SET_PERMISSION: {
+      return { ...state, permission: action.value }
+    }
+    case ACTION_TYPES.SET_SCAN: {
+      return { ...state, scan: action.value }
+    }
+    default: {
+      return state
+    }
+  }
+}
+
+export default function ScanNewSecret() {
+  const [{ scan, permission }, dispatch] = useReducer(scanReducer, initialState)
+  const { user } = useAuthenticationContext()
+  const { add } = useSecretsContext()
```

Comment:
nit: these hooks could be named without the Context suffix. We don't really care that these give you access to a React context, we only care that we get what we need out of them, which are auth and secrets utilities, respectively